shim_maker.py

```diff
@@ -21,62 +26,138 @@ def fancy_list(prefix: str, to_join: Iterable[str], limit: int = 100, with_brack
     return prefix + "(\n    " + ",\n    ".join(to_join) + ",\n)"
 
 
-def create_imports(module_name: str) -> str:
+def sort_imports(imports: str) -> str:
+    return isort.api.sort_code_string(imports, file_path=ISORT_CONFIG)
+
+
+def create_file(module_name: str) -> Tuple[Optional[str], str]:
     subpckg = import_module(module_name)
     root_module_name = module_name.split(".")[0] + "."
-    members = [
-        memb for memb, val in inspect.getmembers(subpckg)
+    docstring = None
+    members = []
+    for memb, val in inspect.getmembers(subpckg):
+        if memb == "__doc__":
+            docstring = val
+            continue
+
         if (
             not memb.startswith("__")
             and memb not in ("TYPE_CHECKING",)
-            and (not inspect.ismodule(val) or val.__name__.startswith(root_module_name))  # skip imported external modules
-            and getattr(val, "__module__", module_name).startswith(root_module_name)  # skip types imported from external modules
-        )
-    ]
-    imports = fancy_list(f"from {module_name} import ", members)
+            and (
+                not inspect.ismodule(val) or val.__name__.startswith(root_module_name)
+            )  # skip imported external modules
+            and getattr(val, "__module__", module_name).startswith(
+                root_module_name
+            )  # skip types imported from external modules
+        ):
+            members.append(memb)
+    code = fancy_list(f"from {module_name} import ", members)
 
     try:
-        public_members = [f"\"{memb}\"" for memb in subpckg.__all__]
+        public_members = [f'"{memb}"' for memb in subpckg.__all__]
     except AttributeError:
         public_members = None
 
     if public_members:
         new_all = fancy_list("__all__ = ", public_members, with_brackets=True)
-        imports += "\n\n" + new_all
-
-    imports += f"\n\nfrom {module_name} import __dict__ as __original_dict__\nlocals().update(__original_dict__)"
-
-    return imports + "\n"
-
-
-def shim_folder(path: str, pypath: str, shim_path: str) -> None:
-    for fn in os.listdir(path):
-        if fn in ("__init__.py", "__main__.py", "py.typed"):
-            with open(f"{path}/{fn}", "r", encoding="utf-8") as f:
-                content = f.read()
-            with open(f"{shim_path}/{fn}", "w", encoding="utf-8") as f:
-                f.write(content)
-
-        elif fn.startswith("__"):
-            pass
+        code += "\n\n" + new_all
+
+    code += f"\n\n# isort: split\nfrom {module_name} import __dict__ as __original_dict__\nlocals().update(__original_dict__)"
+
+    return docstring, sort_imports(code)
+
+
+def shim_folder(path: pathlib.Path, pypath: str, shim_path: pathlib.Path) -> List[pathlib.Path]:
+    modified = []
+    for fn in path.iterdir():
+        mod = fn.relative_to(path)
```

Comment:
```suggestion
        shim_mod = shim_path / fn.relative_to(path)
```

and replace all `shim_path / mod` with `shim_mod` below, as well as `mod` with `fn`